src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/HtmlFragmentCacheContext.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Encodings.Web;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers.Cache
+{
+    /// <summary>
+    /// Context object for html fragment caching.
+    /// </summary>
+    public class HtmlFragmentCacheContext
+    {
+        private IDictionary<string, string> _options;
+
+        /// <summary>
+        /// Creates a new <see cref="HtmlFragmentCacheContext"/>.
+        /// </summary>
+        /// <param name="htmlEncoder">The <see cref="HtmlEncoder"/>.</param>
+        public HtmlFragmentCacheContext(HtmlEncoder htmlEncoder)
+        {
+            HtmlEncoder = htmlEncoder;
```

Comment:
Should probably `null` check the `HtmlEncoder`, otherwise it'll explode when used in the `HybridHtmlFragmentCache`